test/schedulers/trigger_scheduler.cpp

```diff
@@ -0,0 +1,113 @@
+#include "detail/common.hpp"
+
+#include <async/concepts.hpp>
+#include <async/continue_on.hpp>
+#include <async/just_result_of.hpp>
+#include <async/schedulers/trigger_manager.hpp>
+#include <async/schedulers/trigger_scheduler.hpp>
+#include <async/start_detached.hpp>
+#include <async/start_on.hpp>
+#include <async/then.hpp>
+
+#include <catch2/catch_test_macros.hpp>
+
+#include <concepts>
+
+TEST_CASE("trigger_scheduler fulfils concept", "[trigger_scheduler]") {
+    static_assert(async::scheduler<async::trigger_scheduler<"name">>);
+}
+
+TEST_CASE("trigger_scheduler sender advertises nothing",
+          "[trigger_scheduler]") {
+    static_assert(
+        async::sender_of<decltype(async::trigger_scheduler<"name">::schedule()),
+                         async::set_value_t()>);
+}
+
+TEST_CASE("sender has the trigger_scheduler as its completion scheduler",
+          "[trigger_scheduler]") {
+    using S = async::trigger_scheduler<"name">;
+    auto s = S::schedule();
+    auto cs =
+        async::get_completion_scheduler<async::set_value_t>(async::get_env(s));
+    static_assert(std::same_as<decltype(cs), S>);
+}
+
+TEST_CASE("trigger_scheduler schedules tasks", "[trigger_scheduler]") {
+    auto s = async::trigger_scheduler<"name">{};
```

Comment:
issue: using one static scheduler for all tests.. potential state entanglement
I think `"name"` (throughout file) needs to be some unique type trick (decltype-of-lambda) to ensure that tests do not interfere with each other.

Reply:
Good call - I've disentangled them by using `TEMPLATE_TEST_CASE`s